src/main/java/org/spongepowered/common/data/processor/multi/entity/AgeableDataProcessor.java

```diff
@@ -0,0 +1,123 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.multi.entity;
+
+import static org.spongepowered.common.data.util.DataUtil.getData;
+
+import com.google.common.collect.ImmutableMap;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityAgeable;
+import net.minecraft.entity.monster.EntityZombie;
+
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableAgeableData;
+import org.spongepowered.api.data.manipulator.mutable.entity.AgeableData;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeAgeableData;
+import org.spongepowered.common.data.processor.common.AbstractEntityDataProcessor;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class AgeableDataProcessor extends AbstractEntityDataProcessor<Entity, AgeableData, ImmutableAgeableData> {
+
+    public AgeableDataProcessor() {
+        super(Entity.class);
+    }
+
+    @Override
+    protected AgeableData createManipulator() {
+        return new SpongeAgeableData(0, true);
+    }
+
+    @Override
+    protected boolean doesDataExist(Entity entity) {
+        return entity instanceof EntityZombie || entity instanceof EntityAgeable;
+    }
+
+    @Override
+    protected boolean set(Entity entityCast, Map<Key<?>, Object> keyValues) {
+        Integer age = (Integer) keyValues.get(Keys.AGE);
+        boolean adult = (Boolean) keyValues.get(Keys.IS_ADULT);
+
+        if (entityCast instanceof EntityAgeable) {
+            EntityAgeable entity = (EntityAgeable) entityCast;
+
+            if (age != null) {
+                entity.setGrowingAge(age);
+                return true;
+            }
+
+            if (adult) {
+                entity.setGrowingAge(Integer.MAX_VALUE);
+            } else {
+                entity.setGrowingAge(Integer.MIN_VALUE);
+            }
+            return true;
+        } else if (entityCast instanceof EntityZombie) {
+            EntityZombie entity = (EntityZombie) entityCast;
+            entity.setChild(!adult);
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    protected Map<Key<?>, ?> getValues(Entity entityCast) {
+        int age;
+        boolean adult;
+
+        if (entityCast instanceof EntityAgeable) {
+            EntityAgeable entity = (EntityAgeable) entityCast;
+            age = entity.getAge();
+            adult = !entity.isChild();
+        } else /*if (entityCast instanceof EntityZombie)*/ { // commented so it doesn't throw an error
```

Comment:
so initialize them? or throw an error in an else that the entity is unsupported.